clang-tools-extra/clangd/unittests/tweaks/DefineOutlineTests.cpp

```diff
@@ -349,6 +358,36 @@ TEST_F(DefineOutlineTest, ApplyTest) {
   }
 }
 
+TEST_F(DefineOutlineTest, InCppFile) {
+  FileName = "Test.cpp";
+
+  struct {
+    llvm::StringRef Test;
+    llvm::StringRef ExpectedSource;
+  } Cases[] = {
+      {
+          R"cpp(
+            namespace foo {
+            namespace {
+            struct Foo { void ba^r() {} };
```

Comment:
might be worth also inserting some extra declarations in between to see how insertion point selection works

Reply:
> so what about changing this logic to get to enclosing decl instead (i.e. TagDecl) and insert right after its end location?

That will break with nested classes, won't it?

Reply:
Ah, maybe getOuterLexicalRecordContext().

Reply:
Hm, the SourceLocations are weird: getEndLoc() points to the closing brace of the class. Do I really have to look for the semicolon manually?